Dynatrace/dynatrace-operator
# Propagate metadata annotations from pod to metadata props files

## Description
[K8S-10838](https://dt-rnd.atlassian.net/browse/K8S-10838)

Basic idea:
Every annotation that:
- Has the `metadata.dynatrace.com/` prefix in it's key
- Is on the injected **pod**
   - annotations propagated by use via the EnrichmentRules also count (so select annotations from the namespace)
should be written into the `/var/lib/dynatrace/enrichment/dt_metadata.properties` and `/var/lib/dynatrace/enrichment/dt_metadata.json` files on each container of the injected pod.
- in the files, the key should no longer have the metadata prefix.

Workflow:
1. Propagate the annotations from namespace to pod (according to EnrichmentRules)
2. Propagate all annotations with metadata prefix from the pod to init-container envVar (as json)
3. Use the envVar's contents during the creation of the `/var/lib/dynatrace/enrichment/dt_metadata.*` files

## How can this be tested?
Same as https://github.com/Dynatrace/dynatrace-operator/pull/3386, but now the annotation's in question should also appear in the `/var/lib/dynatrace/enrichment/dt_metadata.*` files. (without the prefix)

An "easier" way to test it is to just put annotations with the metadata prefix on the to-be-injected Pod, and see those get propagated to the files inside the containers.
